integrations/launchdarkly/client.py

```diff
@@ -0,0 +1,215 @@
+from port_ocean.utils import http_async_client
+import httpx
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from enum import StrEnum
+import asyncio
+from port_ocean.context.event import event
+
+PAGE_SIZE = 100
+
+
+class ObjectKind(StrEnum):
+    PROJECT = "project"
+    AUDITLOG = "auditlog"
+    FEATURE_FLAG = "flag"
+    ENVIRONMENT = "environment"
+
+
+class LaunchDarklyClient:
+    def __init__(self, api_token: str, launchdarkly_url: str):
+        self.api_url = f"{launchdarkly_url}/api/v2"
+        self.api_token = api_token
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_token}",
+            "Content-Type": "application/json",
+        }
+
+    async def get_paginated_resource(
+        self, kind: str, resource_path: str | None = None, page_size: int = PAGE_SIZE
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        kind = kind + "s"
+        url = kind if not resource_path else f"{kind}/{resource_path}"
+        url = url.replace("auditlogs", ObjectKind.AUDITLOG)
+        params = {"limit": page_size}
+
+        while url:
+            try:
+                response = await self.send_api_request(
+                    endpoint=url, query_params=params
+                )
+                items = response.get("items", [])
+                logger.info(f"Received {kind} batch with {len(items)} items")
+                yield items
+
+                if "_links" in response and "next" in response["_links"]:
+                    url = response["_links"]["next"]["href"]
+                else:
+                    total_count = response.get("totalCount")
+                    logger.info(f"Fetched {total_count} {kind} from Launchdarkly")
+                    break
+
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+            except httpx.HTTPError as e:
+                logger.error(
+                    f"HTTP error occurred while fetching {kind} from LaunchDarkly: {e}"
+                )
+                raise
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        logger.info(f"Requesting Launchdarkly data for endpoint: {endpoint}")
+        try:
+            endpoint = endpoint.replace("/api/v2/", "")
+            url = f"{self.api_url}/{endpoint}"
+            logger.debug(

```

Comment:
The log above is incorrect because you log that you are about to request data for endpoint with `api/v2` and then a line later you are removing the `api/v2`. This is very confusing and I don't understand why you are removing the `api/v2`

Reply:
`self.api_url = f"{launchdarkly_url}/api/v2"`

Reply:
why you set `self.api_url = f"{launchdarkly_url}/api/v2"` and then remove it?

Reply:
I'm removing /api/v2/ from `endpoint` because it will be concatenated with `self.api_url` which already has /api/v2 in it construction.
Also, endpoint doesn't have the full url for making the request, it lacks the host, hence the need for a concatenation